### PR TITLE
[master][KOGITO-1636] - Add missing test coverage for runtime_assemble

### PR DESCRIPTION
**Pull request by sutaakar**
Also reproduces https://issues.redhat.com/browse/KOGITO-1636

### CODE REVIEW

**Comment by sutaakar** [comment]
The failing test is a reproducer for issue in JIRA.

**Comment by spolti** [comment]
Is this PR depending on https://issues.redhat.com/browse/KOGITO-1636 ?

**Comment by sutaakar** [comment]
@spolti yes

**Comment by spolti** [comment]
ok, thanks.

**Comment by sutaakar** [comment]
waiting for https://issues.redhat.com/browse/KOGITO-1636

File: modules/kogito-s2i-core/tests/bats/s2i-core.bats [hunk 1]
```diff
@@ -127,6 +127,25 @@ teardown() {
     [ "${lines[7]}" = "'./myapp.jar' -> '$KOGITO_HOME/bin/myapp.jar'" ]
 }
 
+@test "test runtime_assemble with binary builds native binary" {
+    mkdir -p ${KOGITO_HOME}/bin
+    # emulating an upload
+    mkdir -p /tmp/src/
```
Comment (spolti) [inline modules/kogito-s2i-core/tests/bats/s2i-core.bats:133]:
```suggestion
    mkdir -p /tmp/src/target
```

File: modules/kogito-s2i-core/tests/bats/s2i-core.bats [hunk 2]
```diff
@@ -127,6 +127,25 @@ teardown() {
...
+    touch /tmp/src/myapp-0.0.1-runner
```
Comment (spolti) [inline modules/kogito-s2i-core/tests/bats/s2i-core.bats]: need to create a valid binary, e.g.:


```suggestion
    echo "
        int     main(){
                return 100;
        }" >  test.c
    gcc test.c
    mv a.out /tmp/src/target/myapp-0.0.1-runner
```

File: modules/kogito-s2i-core/tests/bats/s2i-core.bats [hunk 3]
```diff
@@ -144,6 +163,100 @@ teardown() {
...
+    [ -f "$KOGITO_HOME/bin/lib/mydependency-0.0.1.jar" ]
+}
+
+# Check that the irrelevant binaries are excluded
+@test "test runtime_assemble with binary builds entire target Quarkus native build" {
+    mkdir -p ${KOGITO_HOME}/bin
+    # emulating an upload
+    mkdir -p /tmp/src/target
+    touch /tmp/src/target/myapp-0.0.1.jar
+    touch /tmp/src/target/myapp-0.0.1-runner
```
Comment (spolti) [inline modules/kogito-s2i-core/tests/bats/s2i-core.bats]: same here.


```suggestion
    echo "
        int     main(){
                return 100;
        }" >  test.c
    gcc test.c
    mv a.out /tmp/src/target/myapp-0.0.1-runner
```

**Comment by spolti** [comment]
@sutaakar fix is merged, can you please rebase, apply the suggestions and test again?

File: modules/kogito-s2i-core/tests/bats/s2i-core.bats (hunk 1 above)
Comment (sutaakar) [reply modules/kogito-s2i-core/tests/bats/s2i-core.bats:133]: Actually I simulate here uploading of just the binary file, not whole target dir. Target dir upload with binary file is tested in  "test runtime_assemble with binary builds entire target Quarkus native build".

**Comment by sutaakar** [comment]
@spolti Rebased and updated.

File: modules/kogito-s2i-core/tests/bats/s2i-core.bats (hunk 1 above)
Comment (spolti) [reply modules/kogito-s2i-core/tests/bats/s2i-core.bats:133]: ok, no worries.

File: modules/kogito-s2i-core/tests/bats/s2i-core.bats [hunk 4]
```diff
@@ -127,6 +127,28 @@ teardown() {
...
+                return 100;
+        }" >  test.c
+    gcc test.c
+    mv a.out /tmp/src/myapp-0.0.1-runner
+
+    run runtime_assemble
+
+    echo "result= ${lines[@]}"
+    [ "$status" -eq 0 ]
+    [ "${lines[8]}" = "'./myapp-0.0.1-runner' -> '$KOGITO_HOME/bin/myapp-0.0.1-runner'" ]
```
Comment (spolti) [inline modules/kogito-s2i-core/tests/bats/s2i-core.bats:140]: I think this should changed, (lines[8]) I can see the artifact being copied, but sstill failing.
Also, I didn't see the message on the log: https://github.com/kiegroup/kogito-images/blob/master/modules/kogito-s2i-core/added/s2i-core#L64

**Review by spolti — CHANGES_REQUESTED** [review]

**Comment by spolti** [comment]
do you mind to cherry pick it for 0.9.x too?

thanks

**Comment by sutaakar** [comment]
@spolti Done, there was an issue with Jenkins not being able to build the binary so I stored it in GIT and just copy it when needed.

**Review by spolti — APPROVED** [review]

**Pull request merged**